nservicebus/messaging/message-type-detection.md

```diff
@@ -1,24 +1,24 @@
 ---
 title: Message type detection
 summary: Describes the rules for mapping a transport message to a message type
-reviewed: 2021-06-05
+reviewed: 2023-08-04
 component: Core
+related:
+- samples/consumer-driven-contracts
 ---
 
-In order to invoke the correct message handlers for incoming messages, NServiceBus must be able to map the incoming transport message to a [message type](/nservicebus/messaging/messages-events-commands.md).
+NServiceBus sets the [`NServiceBus.EnclosedMessageTypes` header](/nservicebus/messaging/headers.md#serialization-headers-nservicebus-enclosedmessagetypes) to the [AssemblyQualifiedName](https://docs.microsoft.com/en-us/dotnet/api/system.type.assemblyqualifiedname) of the emitted message type when sending, publishing or replying.
+
+It is possible to change the enclosed message type header for all emitted messages by using a behavior. The [consumder-driven-contracts sample](/samples/consumer-driven-contracts) shows how to manipulate the header for outgoing messages.
+
+To invoke the correct message handlers for incoming messages, NServiceBus must be able to map the incoming transport message to a [message type](/nservicebus/messaging/messages-events-commands.md).
```

Comment:
I think this might make a better introduction at the beginning of the page and I'd suggest to move it up.

Reply:
I originally had it that way but figured that talking first about the producer of the message and what that one adds is better and then lead towards how the information is used and what the behavior is depending on its presence or absence.